presto-hive/src/test/java/com/facebook/presto/hive/TestHiveSkipEmptyFiles.java

```diff
@@ -0,0 +1,392 @@
+        Map<String, String> properties = ImmutableMap.<String, String>builder()
+                .put("hive.metastore", "file")
+                .put("hive.metastore.catalog.dir", catalogDirectory.toFile().toURI().toString())
+                .put("hive.allow-drop-table", "true")
+                .put("hive.non-managed-table-writes-enabled", "true")
+                .put("hive.parquet.use-column-names", "true")
+                .put("hive.compression-codec", "GZIP")
+                .put("hive.storage-format", "PARQUET")
+                .put("hive.bucket-execution", "true")
+                .put("hive.skip-empty-files", "true")
+                .build();
+
+        this.queryBucketRunner.createCatalog(CATALOG, CATALOG, properties);
+        this.queryBucketRunner.execute(format("CREATE SCHEMA %s.%s", CATALOG, SCHEMA));
+    }
+
+    @BeforeClass
+    private void createQueryBucketFailRunner()
+            throws Exception
+    {
+        Session session = testSessionBuilder().setCatalog(CATALOG).setSchema(SCHEMA).setTimeZoneKey(TimeZoneKey.UTC_KEY).build();
+        this.queryBucketFailRunner = DistributedQueryRunner.builder(session).setExtraProperties(ImmutableMap.<String, String>builder().build()).build();
+
+        this.queryBucketFailRunner.installPlugin(new HivePlugin(CATALOG));
+        Path catalogDirectory = this.queryBucketFailRunner.getCoordinator().getDataDirectory().resolve("hive_data").getParent().resolve("catalog");
+        Map<String, String> properties = ImmutableMap.<String, String>builder()
+                .put("hive.metastore", "file")
+                .put("hive.metastore.catalog.dir", catalogDirectory.toFile().toURI().toString())
+                .put("hive.allow-drop-table", "true")
+                .put("hive.non-managed-table-writes-enabled", "true")
+                .put("hive.parquet.use-column-names", "true")
+                .put("hive.compression-codec", "GZIP")
+                .put("hive.storage-format", "PARQUET")
+                .put("hive.bucket-execution", "true")
+                .put("hive.skip-empty-files", "false")
+                .build();
+
+        this.queryBucketFailRunner.createCatalog(CATALOG, CATALOG, properties);
+        this.queryBucketFailRunner.execute(format("CREATE SCHEMA %s.%s", CATALOG, SCHEMA));
+    }
+
+    /**
+     * Generates a temporary directory and creates two parquet files inside, one is empty and the other is not
+     *
+     * @param tableName a {@link String} containing the desired table name
+     * @return a {@link File} pointing to the newly created temporary directory
+     */
+    private static File generateMetadata(String tableName)
+            throws Exception
+    {
+        // obtains the root resource directory in order to create temporary tables
+        URL url = TestHiveSkipEmptyFiles.class.getClassLoader().getResource(".");
```

Comment:
Temporary files should be located with either the Java temp fiels API or TestNG APIs, not the classloader.

Reply:
Fixed